experimental/sdk/webpubsub-socketio-extension/src/SIO/components/web-pubsub-adapter.ts

```diff
@@ -49,9 +63,45 @@ export class WebPubSubAdapterInternal extends NativeInMemoryAdapter {
    * @param packet - the packet object
    * @param opts - the options
    */
-  public override broadcast(packet: Packet, opts: BroadcastOptions): void {
-    // TODO: Implement this method.
-    // For now, it is intentionally empty.
+  public override async broadcast(packet: Packet, opts: BroadcastOptions): Promise<void> {
+    debug(`broadcast packet ${JSON.stringify(packet)}`);
+    try {
+      const flags = opts.flags || {};
+      const packetOpts = {
+        preEncoded: true,
+        volatile: flags.volatile,
+        compress: flags.compress,
+      };
+
+      (packet as any).nsp = this.nsp.name;
+
+      const encodedPackets = (this as any)._encode(packet, packetOpts);
```

Comment:
Can this and other private method be dangerous? Copy maybe better?

Reply:
Native `_encode` is [here](https://github.com/socketio/socket.io-adapter/blob/2.5.2/lib/index.ts#L233), where L236 `if` is always false because `canPreComputeFrame` is designed for WebSocket. So I'll modify this line to 
`const encodedPackets = (this as any).encoder.encode(packet, packetOpts);`

Reply:
Native `_encode` is [here](https://github.com/socketio/socket.io-adapter/blob/2.5.2/lib/index.ts#L233), where L236 if is always false because `canPreComputeFrame` is designed for WebSocket. So I'll modify this line to
`const encodedPackets = this._encoder.encode(packet, packetOpts);`

`this._encoder` is fixed to default encoder